src/Elastic.Apm.Azure.ServiceBus/AzureMessagingServiceBusDiagnosticListener.cs

```diff
@@ -0,0 +1,246 @@
+		{
+			_realAgent = agent as ApmAgent;
+			_service = Service.GetDefaultService(agent.ConfigurationReader, agent.Logger);
+			_service.Framework = new Framework { Name = ServiceBus.SegmentName };
+		}
+
+		protected override void HandleOnNext(KeyValuePair<string, object> kv)
+		{
+			Logger.Trace()?.Log("Called with key: `{DiagnosticEventKey}'", kv.Key);
+
+			if (string.IsNullOrEmpty(kv.Key))
+			{
+				Logger.Trace()?.Log($"Key is {(kv.Key == null ? "null" : "an empty string")} - exiting");
+				return;
+			}
+
+			switch (kv.Key)
+			{
+				case "ServiceBusSender.Send.Start":
+					OnSendStart(kv, "SEND");
+					break;
+				case "ServiceBusSender.Schedule.Start":
+					OnSendStart(kv, "SCHEDULE");
+					break;
+				case "ServiceBusReceiver.Receive.Start":
+					OnReceiveStart(kv, "RECEIVE");
+					break;
+				case "ServiceBusReceiver.ReceiveDeferred.Start":
+					OnReceiveStart(kv, "RECEIVEDEFERRED");
+					break;
+				case "ServiceBusSender.Send.Stop":
+				case "ServiceBusSender.Schedule.Stop":
+				case "ServiceBusReceiver.Receive.Stop":
+				case "ServiceBusReceiver.ReceiveDeferred.Stop":
+					OnStop();
+					break;
+				case "ServiceBusSender.Send.Exception":
+				case "ServiceBusSender.Schedule.Exception":
+				case "ServiceBusReceiver.Receive.Exception":
+				case "ServiceBusReceiver.ReceiveDeferred.Exception":
+					OnException(kv);
+					break;
+				default:
+					Logger.Trace()?.Log("`{DiagnosticEventKey}' key is not a traced diagnostic event", kv.Key);
+					break;
+			}
+		}
+
+		private void OnReceiveStart(KeyValuePair<string, object> kv, string action)
+		{
+			if (!(kv.Value is Activity activity))
+			{
+				Logger.Trace()?.Log("Value is not an activity - exiting");
+				return;
+			}
+
+			string queueName = null;
+			foreach (var tag in activity.Tags)
+			{
+				switch (tag.Key)
+				{
+					case "message_bus.destination":
+						queueName = tag.Value;
+						break;
+					default:
+						continue;
+				}
+			}
+
+			if (MatchesIgnoreMessageQueues(queueName))
+				return;
+
+			var transactionName = queueName is null
+				? $"{ServiceBus.SegmentName} {action}"
+				: $"{ServiceBus.SegmentName} {action} from {queueName}";
+
+			var transaction = ApmAgent.Tracer.StartTransaction(transactionName, ServiceBus.Type);
+			transaction.Context.Service = _service;
+
+			// transaction creation will create an activity, so use this as the key.
+			var activityId = Activity.Current.Id;
+
+			if (!_processingSegments.TryAdd(activityId, transaction))
+			{
+				Logger.Error()?.Log(
+					"Could not add {Action} transaction {TransactionId} for activity {ActivityId} to tracked segments",
+					action,
+					transaction.Id,
+					activityId);
+			}
+		}
+
+		private bool MatchesIgnoreMessageQueues(string name)
+		{
+			if (name != null && _realAgent != null)
+			{
+				var matcher = WildcardMatcher.AnyMatch(_realAgent.ConfigStore.CurrentSnapshot.IgnoreMessageQueues, name);

```

Comment:
I didn't try with central configuration. I was assuming that since this is part of the merged messaging spec, it should be part of the implementation. I'll add a test for it.